Analytics/Classes/Internal/SEGHTTPClient.h

```diff
@@ -15,6 +15,8 @@ NS_ASSUME_NONNULL_BEGIN
 @interface SEGHTTPClient : NSObject
 
 @property (nonatomic, strong) SEGRequestFactory requestFactory;
+@property (nonatomic, readonly) NSMutableDictionary<NSString *, NSURLSession *> *sessionsByWriteKey;
```

Comment:
Just seems like an unnecessary extra step. Even if the http client was being used by multiple writeKeys, why wouldn't you want to reuse the sessions between the writeKeys?